snapshottest/module.py

```diff
@@ -260,11 +260,17 @@ def save_changes(self):
     def remove_fields(cls, input, remove_fields_list=None):
         if remove_fields_list is None:
             remove_fields_list = []
-        gen = (field for field in remove_fields_list if field in input)
-        for field in gen:
-            del input[field]
+
+        if isinstance(input, list):
+            for el in input:
+                cls.remove_fields(el, remove_fields_list)
+
         if isinstance(input, dict):
-            gen = (value for value in input.values() if isinstance(value, dict))
+            gen = (field for field in remove_fields_list if field in input)
```

Comment:
does this work for arbitrarily nested dicts? Should add some tests for that

Reply:
To confirm, did you mean supporting nested dicts, like `{{}, {}}`? If so, do we need to support that? I am working under the assumption that snapshots will be valid JSON, which the above is not.

Reply:
```{
    "test": {
        "another_one": {
            "u smaht": {
                "dj khaled": {
                    "ft. drake": {
                        "yes": "no"
                    }
                }
            }
        }
    }
}```